sub-packages/bionemo-testing/src/bionemo/testing/testing_callbacks.py

```diff
@@ -14,171 +14,239 @@
-    """
-
-    def __init__(
-        self, metadata_path: pathlib.Path, metrics_getter: dict[str, Callable[[pl.Trainer, pl.LightningModule], Any]]
-    ):
-        """Initialises callback with path and called information.
-
-        Args:
-            metadata_path (pathlib.Path): Path where the metadata will be saved.
-            metrics_getter (dict[str, Callable[[pl.Trainer, pl.LightningModule], Any]]): A dictionary of metadata keys and their corresponding functions. Must be a subset of the dictionary passed to MetadataSaveCallback.
+            User must override get_metadata to get metadata from the trainer and pl_module.
         """
-        self.metadata_path = metadata_path
-        self.metrics_getter = metrics_getter
-
-    def on_train_start(self, trainer: Trainer, pl_module: LightningModule):
-        """Loads associated metadata and compares with current metrics."""
-        pickle_file_path = os.path.join(self.metadata_path, "checkpoints/metadata.pkl")
-        # check that pickle file exists
-        assert os.path.isfile(pickle_file_path), f"No file found at {pickle_file_path}"
-        with open(pickle_file_path, "rb") as metadata_file:
-            metadata_dict = pickle.load(metadata_file)
-        current_metadata = {}
-        for metadata_key, func in self.metrics_getter.items():
-            expected_value = metadata_dict[metadata_key]
-            current_value = func(trainer, pl_module)
-            current_metadata[metadata_key] = current_value
-
-        # TODO (SKH): Ideally this would collect _all_ failures instead of failing on the first one.
-        for metadata_key in current_metadata:
-            expected_value = metadata_dict[metadata_key]
-            current_value = current_metadata[metadata_key]
-            assert (
-                expected_value == current_value
-            ), f"Value mismatch for key {metadata_key}: stored_value={expected_value}, current_value={current_value}"
-        # Cleanup
-        os.remove(pickle_file_path)
+        if mode not in [Mode.STOP, Mode.RESUME]:
+            raise ValueError(f"mode must be 'stop' or 'go', got {mode}")
+        self.mode = mode
+        super().__init__()
+
+    @abstractmethod
+    def get_metadata(self, trainer: Trainer, pl_module: LightningModule) -> Any:
+        """Get metadata from trainer and pl_module."""
+        raise NotImplementedError
+
+    def on_train_epoch_start(self, trainer: Trainer, pl_module: LightningModule):  # noqa: D102
+        if self.mode == Mode.RESUME:
+            self.data = self.get_metadata(trainer, pl_module)
+
+    def on_validation_epoch_end(self, trainer: Trainer, pl_module: LightningModule):  # noqa: D102
+        if not trainer.sanity_checking and self.mode == Mode.STOP:
+            self.data = self.get_metadata(trainer, pl_module)
+
+
+class TrainValInitConsumedSamplesStopAndGoCallback(AbstractStopAndGoCallback):
```

Comment:
@pstjohn Can you highlight why we aren't able to refactor this to use your new base class?

Reply:
done